lib/crc/crc8_sw.c

```diff
@@ -7,15 +7,15 @@
 
 #include <crc8.h>
 
-static u8_t crc8_ccitt_small_table[16] = {
+static const u8_t crc8_ccitt_small_table[16] = {
 	0x00, 0x07, 0x0e, 0x09, 0x1c, 0x1b, 0x12, 0x15,
 	0x38, 0x3f, 0x36, 0x31, 0x24, 0x23, 0x2a, 0x2d
 };
 
 u8_t crc8_ccitt(u8_t val, const void *buf, size_t cnt)
 {
 	int i;
-	const u8_t *p = buf;
+	const u8_t *p = (const u8_t *)buf;
```

Comment:
This cast is superfluous (and noisy) in C, we don't use such in Zephyr, so it looks out of place here too.